### ISSUE TITLE
I can't connect neo4j instance, need to see all databases

### ISSUE THREAD

**Issue by nuhyurduseven** [neo4j/neo4j-jdbc#461]
## Expected Behavior
connecting to neo4j instance, need to see all dbs

## Actual Behavior
connected to `neo4j` database without specify in connection string

## How to Reproduce the Problem
I want to connect neo4j instance with jdbc driver and see all database but I see only `neo4j` database. I am using dbeaver as database client. When I use  `jdbc:neo4j:neo4j:<host:port>` , I can not see all neo4j database. 

### Versions

  - OS: ubuntu 20.04
  - Library: Neo4j JDBC Driver 4.0.9  & bi-connector jar   Neo4jJDBC 1.0.10.1011
  - Neo4j: Neo4j 4.4.23

![image](https://github.com/neo4j-contrib/neo4j-jdbc/assets/138458266/036d746d-cbe1-401c-adc9-4121d5b171e6)

**Comment by fbiville** [comment]
@nuhyurduseven indeed, `java.sql.DatabaseMetaData#getCatalogs` is currently implemented to return an empty list.
Would you like to submit a PR to change `org.neo4j.jdbc.bolt.BoltNeo4jDatabaseMetaData` so that it returns the result of the Cypher query `SHOW DATABASES`?

**Comment by nuhyurduseven** [comment]
Yes, I need to do this feature. But I don't use `bolt`. I used `neo4j` instead. I can send a feature request for this

**Comment by fbiville** [comment]
`Bolt` in the classname simply refers to the Bolt protocol, which "materializes" as the `bolt` and `neo4j` schemes in the connection URIs.

**Comment by michael-simons** [comment]
This is implemented with https://github.com/neo4j/neo4j-jdbc/commit/d6b2183c62a943ab5b074e062f3e7d427acdbcb3 and works in an informative fashion:

![image](https://github.com/user-attachments/assets/b4f12f71-332d-424f-acd5-19dc40fdf69b)

It will be of limited use for now though, as the underlying bolt implementation is not capable of switching the database (catalog) as of now and I created a follow up for this here: #820
